### ISSUE TITLE
query_string does not respect default_operator when a wildcard query produces multiple tokens

### ISSUE THREAD

**Issue by lukapor** [elastic/elasticsearch#17649]
query string does not return right hits when you use query with wildcard and it is tokenized. Example

 "query_string": {
                  "query": "luka-por-si*",
                  "default_operator": "and",
                  "allow_leading_wildcard": false,
                  "analyze_wildcard": true,
                  "minimum_should_match": "100%",
                  "fields": [
                    "name"
                  ],
                  "use_dis_max": false
                }

query should return hits with luka AND por AND si*. On es version < 2.3 that works ok, now on version 2.3 and 2.3.1 query returns luka OR por OR si*. I attached files to reproduce this scenario. 

**Elasticsearch version**:>=2.3

**JVM version**:8

**OS version**:any (tested on win and Linux ubuntu)

**Steps to reproduce**:
 1. create index (createindexmapping.txt)
 2. fill data (insertdata.txt)
 3. get results (searchdata.txt)
[createindexmapping.txt](https://github.com/elastic/elasticsearch/files/212986/createindexmapping.txt)
[insertdata.txt](https://github.com/elastic/elasticsearch/files/212988/insertdata.txt)
[searchdata.txt](https://github.com/elastic/elasticsearch/files/212987/searchdata.txt)

**Comment by jimferenczi** [comment]
@lukapor the behavior is the same for 2.3 and <= 2.3. The behavior is described here: https://github.com/elastic/elasticsearch/issues/1539
Though I agree that the query_string parser should respect the default operator when it deals with prefix query on analyzed fields. Could you please rename your issue with something like:
"query_string does not respect default_operator when a wildcard query produces multiple tokens"

**Comment by clintongormley** [comment]
Duplicate of https://github.com/elastic/elasticsearch/issues/2183

**Comment by lukapor** [comment]
@clintongormley I don't think that is duplicate issue.
Attached example returns on es version 2.2.0 (< 2.3.0) only one document, on lastest es version (2.3.0, 2.3.1) return all (3) documents.

**Comment by clintongormley** [comment]
OK - it seems this change in behaviour is due to https://github.com/elastic/elasticsearch/pull/16155

The explanation in 2.3 is:

    "explanation": "(+(+(name:luka* name:por* name:si*)))"

while in 2.2 it is:

    "explanation": "(+(+(name:luka* name:por* name:si*)))~3"

This happens even if you remove the `minimum_should_match` parameter.

I'm not sure what the correct behaviour should be in this situation.  @jimferenczi what do you think?

**Comment by jimferenczi** [comment]
Yes this is a side effect of the way we handle analyzed wildcard. In 2.x coords are disabled on analyzed wildcard query that produces multiple tokens. I did not check this behavior when I've merged https://github.com/elastic/elasticsearch/pull/16155. 
IMO the 2.3 and 2.2 query are wrong and the real fix is in the master branch: https://github.com/elastic/elasticsearch/pull/17711.
... which produces:
````
+(+(+name:luka +name:por +name:si*))
````
... and with "default_operator": "or":
````
+(+((name:luka name:por name:si*)~3))
````

I can backport this fix to 2.x if needed, @clintongormley ?

**Comment by clintongormley** [comment]
yes please @jimferenczi

**Comment by lukapor** [comment]
Nice @jimferenczi 

@clintongormley have you any clue, when it will be the next release with this patch?

**Comment by clintongormley** [comment]
Closed by #17711
